src/metabase/util/schema.clj

```diff
@@ -198,58 +212,108 @@
 ;;; |                                                 USEFUL SCHEMAS                                                 |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(spec/def ::defschema-args
+  (spec/cat
+   :name   symbol?
+   :docstr (spec/? string?)
+   :body   (spec/* any?)))
+
+(defmacro defschema
+  "Define a Malli schema"
+  [& args]
+  (let [{:keys [name docstr body]} (spec/conform ::defschema-args args)]
+    `(def ~(vary-meta name assoc :doc docstr) (mc/schema ~@body))))
+
 (def NonBlankString
   "Schema for a string that cannot be blank."
   (with-api-error-message (s/constrained s/Str (complement str/blank?) "Non-blank string")
     (deferred-tru "value must be a non-blank string.")))
 
+(defschema NonBlankStringMalli
+  "Schema for a string that cannot be blank."
+  [:string {:min 1}])
+
 (def IntGreaterThanOrEqualToZero
   "Schema representing an integer than must also be greater than or equal to zero."
   (with-api-error-message
     (s/constrained s/Int (partial <= 0) (deferred-tru "Integer greater than or equal to zero"))
     (deferred-tru "value must be an integer greater than or equal to zero.")))
 
+(defschema IntGreaterThanOrEqualToZeroMalli
+  "Schema representing an integer than must also be greater than or equal to zero."
+  [:int {:min 0}])
+
 ;; TODO - rename this to `PositiveInt`?
 (def IntGreaterThanZero
   "Schema representing an integer than must also be greater than zero."
   (with-api-error-message
     (s/constrained s/Int (partial < 0) (deferred-tru "Integer greater than zero"))
     (deferred-tru "value must be an integer greater than zero.")))
 
-(def NonNegativeInt
```

Comment:
this is exactly the same as `IntGreaterThanOrEqualToZero`. so I replaced all the places that use it with `IntGreaterThanOrEqualToZero`.